src/custom-elements/src/components/progress/progress.tsx

```diff
@@ -0,0 +1,52 @@
+import { Component, Prop, Watch } from '@stencil/core';
+
+@Component({
+  tag: 'chi-progress',
+  styleUrl: 'progress.scss',
+  scoped: true
+})
+export class Progress {
+  /**
+   *  to set the value of a progress bar.
+   */
+  @Prop({ reflectToAttr: true }) value: string;
+
+  /**
+   *  to set the min value of a progress bar.
+   */
+  @Prop({ reflectToAttr: true }) min = '0';
+
+  /**
+   *  to set the max value of a progress bar.
+   */
+  @Prop({ reflectToAttr: true }) max = '100';
+
+  /**
+   *  to indicate a progress bar state { info, warning, danger, success }.
+   */
+  @Prop({ reflectToAttr: true }) state = 'success';
+
+  @Watch('state')
+  stateValidation(newValue: string) {
+    if (newValue && !['info', 'warning', 'danger', 'success'].includes(newValue)) {
+      throw new Error(`${newValue} is not a valid state for progress. If provided, valid values are: info, warning, danger or success. `);
```

Comment:
What do you think about expanding this message with: "_; if not, 'success' value will be set._"

Reply:
@SergioQCL the docs already mention that success is the default state, but It's a good idea I'll update that.